pkg/operation/botanist/monitoring.go

```diff
@@ -69,7 +69,7 @@ func observabilityIngressSecretConfig(name string) *secrets.BasicAuthSecretConfi
 // DeploySeedMonitoring installs the Helm release "seed-monitoring" in the Seed clusters. It comprises components
 // to monitor the Shoot cluster whose control plane runs in the Seed cluster.
 func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
-	if b.Shoot.Purpose == gardencorev1beta1.ShootPurposeTesting {
+	if b.Shoot.Purpose == gardencorev1beta1.ShootPurposeTesting || !gardenlethelper.IsMonitoringEnabled(b.Config) {
```

Comment:
You can use your new function here as well :)
```suggestion
	if !b.IsShootMonitoringEnabled() {
```